### PR TITLE
DOC Changelog CMS search with elemental

### PR DESCRIPTION
**Pull request by emteknetnz**
Issue https://github.com/silverstripe/silverstripe-elemental/issues/1067

### CODE REVIEW

File: en/04_Changelogs/5.1.0.md [hunk 1]
```diff
@@ -48,6 +49,16 @@ Note that even if you store the schema in the cache directory, the schema still
 
 See [choosing where the cache is stored](/developer_guides/graphql/getting_started/building_the_schema/#cache-location) for more information about configuring the schema storage directory.
 
+### CMS page search performance
```
Comment (maxime-rainville) [inline en/04_Changelogs/5.1.0.md]:
```suggestion
### Improvement to page search performance with Elemental
```

File: en/04_Changelogs/5.1.0.md [hunk 2]
```diff
@@ -48,6 +49,16 @@ Note that even if you store the schema in the cache directory, the schema still
...
+
+- The CMS search has been optimised to reduce the number of database queries made when searching for pages when the site uses elemental content blocks. This has resulted in a small performance improvement.
```
Comment (maxime-rainville) [inline en/04_Changelogs/5.1.0.md]: Can we give a generic baseline here? 

Something like "In our test environment, with X number of pages we observed a reduction of Y% number of queries and of z% response time. Performance will vary with your environment."

File: en/04_Changelogs/5.1.0.md [hunk 3]
```diff
@@ -48,6 +49,16 @@ Note that even if you store the schema in the cache directory, the schema still
...
+- A new opt-in behaviour is available that makes a very large different to the performance when using elemental content blocks. In testing this more than doubled performance. The opt-in disables the default behaviour of rendering all content blocks for CMS search, and instead simply extracts the database contents of the elements. Note this does not use the `$searchable_fields` config. To opt-in to this behaviour, use the following config:
```
Comment (maxime-rainville) [inline en/04_Changelogs/5.1.0.md]: If they are any downside to the behaviour, can we make them explicit?

File: en/04_Changelogs/5.1.0.md (hunk 2 above)
Comment (maxime-rainville) [inline en/04_Changelogs/5.1.0.md]:
```suggestion
- The CMS search has been optimised to reduce the number of database queries made when searching for pages with elemental content blocks. This has resulted in a small performance improvement.
```

File: en/04_Changelogs/5.1.0.md (hunk 3 above)
Comment (maxime-rainville) [inline en/04_Changelogs/5.1.0.md]:
```suggestion
- A new opt-in behaviour is available that makes a very large difference to the performance when using elemental content blocks. In testing, this behaviour more than halve the response time of SiteTree search request. The opt-in disables the default behaviour of rendering all content blocks for CMS search. Instead, it simply extracts the database contents of the elements. Note this does not use the `$searchable_fields` config. To opt-in to this behaviour, use the following config:
```

**Review by maxime-rainville — CHANGES_REQUESTED** [review]

File: en/04_Changelogs/5.1.0.md (hunk 1 above)
Comment (emteknetnz) [reply en/04_Changelogs/5.1.0.md]: Updated

File: en/04_Changelogs/5.1.0.md (hunk 2 above)
Comment (emteknetnz) [reply en/04_Changelogs/5.1.0.md]: Updated

Comment (emteknetnz) [reply en/04_Changelogs/5.1.0.md]: Updated

File: en/04_Changelogs/5.1.0.md (hunk 3 above)
Comment (emteknetnz) [reply en/04_Changelogs/5.1.0.md]: Have added `There is a downside to consider which is that any related content not directly on the element will not be matched against the search query.`

Comment (emteknetnz) [reply en/04_Changelogs/5.1.0.md]: Updated

**Review by GuySartorelli — APPROVED** [review]
LGTM - will merge after the main PR

File: en/04_Changelogs/5.1.0.md [hunk 4]
```diff
@@ -48,6 +49,25 @@ Note that even if you store the schema in the cache directory, the schema still
...
 
+### Improvement to page search performance with Elemental {#cms-search-performance}
+
+- The CMS search has been optimised to reduce the number of database queries made when searching for pages with elemental content blocks. This has resulted in a small performance improvement. In our test environment, with 1,000 pages each with 5 content blocks we observed a 9% performance improvement. Performance will vary with your environment.
+- A new opt-in behaviour is available that makes a very large difference to performance when using elemental content blocks. In testing, this behaviour more than halved the response time of the sitetree search request. The opt-in disables the default behaviour of rendering all content blocks for CMS search. Instead, it simply extracts the database contents of the elements. There is a downside to consider which is that any related content not directly on the element will not be matched against the search query. Note this does not use the `$searchable_fields` config. To opt-in to this behaviour, use the following config:
```
Comment (michalkleiner) [inline en/04_Changelogs/5.1.0.md]:
```suggestion
- A new opt-in behaviour is available that makes a very large difference to performance when using elemental content blocks. In testing, this behaviour more than halved the response time of the sitetree search request. The opt-in feature disables the default behaviour of rendering all content blocks for CMS search. Instead, it simply extracts the database contents of the elements from its text and html fields. There is a downside to consider which is that any related content not directly on the element will not be matched against the search query. Note this does not use the `$searchable_fields` config. To opt-in to this behaviour, use the following config:
```

Comment (emteknetnz) [reply en/04_Changelogs/5.1.0.md]: Updated

**Pull request merged**